scripts/full-workflow.sh

```diff
@@ -0,0 +1,363 @@
+    cp "$SCRIPTDIR"/test-data/e2e-qna-freeform-skill.yaml "$DATA_HOME"/instructlab/taxonomy/compositional_skills/extraction/inference/qualitative/e2e-siblings/qna.yaml
+
+    # add grounded skill to the taxomony
+    mkdir -p "$DATA_HOME"/instructlab/taxonomy/compositional_skills/extraction/answerability/e2e-yes_or_no
+    cp "$SCRIPTDIR"/test-data/e2e-qna-grounded-skill.yaml "$DATA_HOME"/instructlab/taxonomy/compositional_skills/extraction/answerability/e2e-yes_or_no/qna.yaml
+
+    # add knowledge to the taxomony
+    mkdir -p "$DATA_HOME"/instructlab/taxonomy/knowledge/tonsils/overview/e2e-tonsils
+    cp "$SCRIPTDIR"/test-data/e2e-qna-knowledge.yaml "$DATA_HOME"/instructlab/taxonomy/knowledge/tonsils/overview/e2e-tonsils/qna.yaml
+
+    step Taxonomy Verification
+    ilab taxonomy diff
+    task Taxonomy Updated & Taxonomy Verification complete
+}
+
+test_generate() {
+    task Generate synthetic data
+    ilab data generate
+    task Synthetic Data Generation Complete
+}
+
+test_train() {
+    task Train the model
+    local knowledge_data_path
+    knowledge_data_path=$(find "${DATA_HOME}"/instructlab/datasets -name 'knowledge_train_msgs*' | head -n 1)
+    ilab model train --data-path "${knowledge_data_path}"
+    TRAINED_MODEL_PATH=$(find "${DATA_HOME}"/instructlab/checkpoints/hf_format/ -name 'samples_*' | head -n 1)
+    task Model Training Complete
+}
+
+test_phased_train() {
+    task Train the model with LAB multi-phase training
+
+    # linter wants data_path and other variables declared then assigned separately.
+    # 'Declare and assign separately to avoid masking return values' <-- error.
+    local data_path
+    data_path=$(find "${DATA_HOME}"/instructlab/datasets -name 'messages_*' | head -n 1)
+
+    local model_path
+    model_path="${CACHE_HOME}/instructlab/models/${GRANITE_SAFETENSOR_REPO}"
+
+    # general training args - not the same as the global TRAIN_ARGS
+    local train_args
+    # general phased training args
+    train_args+=(
+        "--phased-training"
+        "--skip-user-confirm"
+        "--phased-mt-bench-judge-dir=${model_path}" # uses base model
+    )
+
+    # phase 1 args 
+    train_args+=(
+        "--phased-phase1-data=${data_path}"
+    )
+
+    # phase 2 args 
+    train_args+=(
+        "--phased-phase2-data=${data_path}"
+    )
+
+    ilab model train "${train_args[@]}"
+    # final best model is written to stdout.
+    task Multi-phase Training Complete
+}
+
+test_evaluate() {
+    task Evaluate the model
+    
+    local model_path
+    local base_model_path
+    local judge_model_path
+    local mmlu_branch_tasks_dir
+
+    model_path=${TRAINED_MODEL_PATH}
+    base_model_path="${CACHE_HOME}/instructlab/models/instructlab/granite-7b-lab"
+    judge_model_path="${CACHE_HOME}/instructlab/models/${PROMETHEUS_7B_MODEL}"
+    mmlu_branch_tasks_dir=$(find "${DATA_HOME}"/instructlab/datasets -name 'node_datasets_*' | head -n 1)
+
+    export INSTRUCTLAB_EVAL_MMLU_MIN_TASKS=true
```

Comment:
Do we want to do full eval here? Maybe we do min tasks for the medium case since it will eventually run on PRs and do full eval on the large case we are planning?